lib/auth.ts

```diff
@@ -24,7 +25,8 @@ export interface GetServerSidePropsAuthContext extends GetServerSidePropsContext
 export function requireAuthentication(
   innerFunction: (
     ctx: GetServerSidePropsAuthContext
-  ) => Promise<GetServerSidePropsResult<Record<string, unknown>>>
+  ) => Promise<GetServerSidePropsResult<Record<string, unknown>>>,
+  authRole: string
```

Comment:
As with the other comments we can also use the `UserRole` here as the type for `authRole`